src/providers/sdks/koboldcpp.ts

```diff
@@ -0,0 +1,97 @@
+import { fetchEventSource } from "@ai-zen/node-fetch-event-source";
+import type { Response as NodeFetchResponse } from "node-fetch";
+
+export interface KoboldInferParams {
+  prompt: string;
+  max_length?: number;
+  top_k?: number;
+  top_p?: number;
+  temperature?: number;
+  tfs?: number;
+  stop_sequence?: string[];
+}
+
+export type OnOpen = (response: NodeFetchResponse) => void | Promise<void>;
+export type OnUpdate = (completion: string) => void | Promise<void>;
+
+const DEFAULT_API_URL = "https://localhost:5001";
+export const DEFAULT_TEMPLATE = "{system}\n\n{prompt}";
+export const DEFAULT_CTX = 2048;
+
+export class Client {
+  private apiUrl: string;
+
+  constructor(private apiKey: string, options?: { apiUrl?: string }) {
+    this.apiUrl = options?.apiUrl ?? DEFAULT_API_URL;
+  }
+
+  /* async complete(params: KoboldInferParams, options?: { signal?: AbortSignal }): Promise<string> {
```

Comment:
I see no reason to use anything else than streamed response: every backend provider supports it, this is a de facto standard. I am in favor of completely remove the `completion` method as it has no value actually and it would simplify the code, which is better for maintenance and implementation of new providers